packages/stream_video/lib/src/sfu/sfu_client.dart

```diff
@@ -37,13 +41,60 @@ class SfuClient {
   final int sessionSeq;
   final String sfuToken;
 
+  int _retryInterval(int numberOfFailures) {
+    // try to reconnect in 0.25-5 seconds (random to spread out the load from failures)
+    final max = (500 + numberOfFailures * 2000).clamp(0, 5000);
+    final min = ((numberOfFailures - 1) * 2000).clamp(250, 5000);
+    return (min + Random().nextDouble() * (max - min)).floor();
+  }
+
+  Future<Result<T>> _executeWithRetry<T>({
+    required Future<T> Function() call,
+    int maxRetries = 3,
+  }) async {
+    var attempt = 0;
+    dynamic dynamicResponse;
+
+    while (attempt < maxRetries) {
+      final response = await call();
+      dynamicResponse = response as dynamic;
+
+      if (dynamicResponse.hasError != null &&
+          dynamicResponse.hasError() &&
+          dynamicResponse.error is sfu_models.Error) {
```

Comment:
Why is this `dynamic`? Looks like it should be `T response`, instead of `dynamic dynamicResponse;`

But you also assume it has methods like `hasError`, so should this be `T extends Result`?